platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/results/serializable/SerializableTestResultStore.java

```diff
@@ -231,8 +212,9 @@ public void completed(TestDescriptorInternal testDescriptor, TestResult testResu
 
             // We remove the id here since no further events should come for this test, and it won't be needed as a parent id anymore
             long id = assignedIds.remove(testDescriptor.getId());
-            resultsEncoder.writeSmallLong(id);
             try {
+                OutputEntry outputEntry = outputWriter.finishOutput(id);
+                OutputEntry.Ser.INSTANCE.write(resultsEncoder, outputEntry);
```

Comment:
💅 This reads really awkward in practice.  I wonder if it would be less awkward to have the constant be a member of `OutputEntry` - something like:
```
private static final Serializer<OutputEntry> SERIALIZER = new Ser();
```
and then you could just reference it as `OutputEntry.SERIALIZER.write(...)`.  I _think_ you could then make `Ser` private, too.  This is pretty minor, so up to you if you think it's worth changing or not.

Reply:
I guess that makes more sense.